extensions/amp-app-banner/1.0/component/android.js

```diff
@@ -0,0 +1,106 @@
+import {docInfoService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+import {xhrService} from '#preact/services/xhr';
+
+import {user} from '#utils/log';
+
+import {openWindowDialog} from '../../../../src/open-window-dialog';
+
+const OPEN_LINK_TIMEOUT = 1500;
+
+/**
+ * @return {{openOrInstall: function(): void, promise: Promise<Response>}|null}
+ */
+export function getAndroidAppInfo() {
+  const win = self.window;
```

Comment:
I think this should only execute on android phones, so maybe using window like this will be ok(?). But I do know we have some issues with using window directly effecting a components ability to render server side. I know there is a utility to get the window object too, which I think is suppose to help protect for SSR issues, maybe it'd be worthwhile to use that?

`import {getWin} from '#core/window';`

Reply:
Good idea; I did find a lot of lint rules disallowing `window` so a common util would be nice.  Will update.

Reply:
Well, I investigated a bit further ... the `getWin` method requires a `node` parameter, which isn't available most places we're using `window` ...  For example, this `getAndroidAppInfo` is used before the component is even mounted, and actually determines _whether_ the component should even mount.  
I also do see usages of raw `window` in at least 5 other Bento components (amp-accordion, amp-facebook, amp-jwplayer, amp-sidebar, amp-social-share) ... 

I'm thinking either we continue using `window` as-is, or we could make it more consistent by putting it behind a `windowService`, which exposes things like `windowService.location` and `windowService.document` etc?

Reply:
We can revisit this later if we find that it's a problem. I assume `getAndroidAppInfo` won't execute unless we're on an android device. So this shouldn't cause a problem on SSR.

Reply:
@dethstrobe what do you think about adding the `windowService` to abstract the window interactions?
One added benefit will be easier testing, too, since it's easier to stub/spy our own service.

Reply:
I don't personally see a lot of value in it.

My slight fear is we write a windowService with a bunch of defensive logic that doesn't need to exist, when we can be somewhat assured that the window or other globals do exist. I think we should make it when we know we need it.

But at the same time, if you think it'd be a worthwhile investment, I am not against it.

Reply:
I don't think it's worthwhile yet.  I was able to do all the testing I needed, using `env.sandbox.stub` to patch even `localStorage` and `window.open`, so I'm happy with that.